ansibullbot/triagers/plugins/docs_info.py

```diff
@@ -0,0 +1,306 @@
+                    )
+
+            for child_node in ast.iter_child_nodes(node):
+                if isinstance(child_node, (ast.FunctionDef, ast.AsyncFunctionDef)):
+                    func_map = ParsedFunc(
+                        name=child_node.name,
+                        line_start=child_node.lineno,
+                        line_end=child_node.end_lineno,
+                    )
+                    func_doc = ast.get_docstring(child_node)
+                    if func_doc is not None:
+                        func_map.doc_string = func_doc
+                        ds_span = re.search(re.escape(func_doc), content)
+                        if ds_span is not None:
+                            func_map.ds_line_start = len(
+                                content[:ds_span.start()].splitlines()
+                            )
+                            func_map.ds_line_end = (
+                                func_map.ds_line_start + len(func_doc.splitlines())
+                        )
+                    class_map.funcs.append(func_map)
+
+            mod_map.classes.append(class_map)
+
+    return mod_map
+
+def _check_py_changes(file_content, diff):
+    """ Check a python file's changes to see if they're only docstring
+        changes.
+    """
+
+    diff = _get_diff_info(diff)
+    source = None
+
+    if file_content is not None:
+        source = _get_ast_info(file_content)
+
+    if source is not None:
+        for line in diff:
+            # Check if change applies to module DOCUMENTATION (if exists)
+            if source.doc_string:
+                if source.ds_line_start <= line["lineno"] <= source.ds_line_end:
+                    continue
+
+            # Check if change applies to module EXAMPLES (if exists)
+            if source.example_string:
+                if source.ex_line_start <= line["lineno"] <= source.ex_line_end:
+                    continue
+
+            # Find appropriate class if it exists
+            source_class = source.find_class(line["lineno"])
+            if source_class is not None:
+                if source_class.doc_string:
+                    if source_class.ds_line_start <= line["lineno"] <= source_class.ds_line_end:
+                        continue
+
+                source_func = source_class.find_function(line["lineno"])
+                if source_func is not None and source_func.doc_string:
+                    if source_func.ds_line_start <= line["lineno"] <= source_func.ds_line_end:
+                        continue
+
+            # If we made it this far, this change is outside docs/examples
+            return False
+        else:
+            return True
+
+def get_docs_facts(iw):
+    """ Cycle through the files and triage if changes only
+        affect docs-related files/content.
+    """
+    dfacts = {
+        "is_docs_only": False
+    }
+
+    if not iw.is_pullrequest():
+        return dfacts
+
+    docs_only = True
```

Comment:
Should we rather start with `docs_only` being set to `False` to prevent false positives in case say `iw.commits` misbehaves and returns an empty list? I have not looked into how likely that is though.

Reply:
Bad habit to shortcut the desired state and avoid a couple lines of code. Having it failsafe to `False` is an excellent point.